datafusion/expr/src/logical_plan/mod.rs

```diff
@@ -0,0 +1,23 @@
+// Licensed to the Apache Software Foundation (ASF) under one
```

Comment:
I wonder if making this a separate crate like `datafusion-logical-plan` might be considered? Is there any reason the logical plan structs need to be in the expr? I could (potentially) see a use of just `Expr`s without also `LogicalPlan`

Reply:
I was working towards having the logical plan and logical expressions in the same create (currently named `datafusion-expr` but potentially could be renamed to `datafusion-logical-plan`).

The main motivation is that I was planning on adding something like:

``` rust
pub enum Expr {
  Exists {
    subquery: Arc<LogicalPlan>,
    negated: bool
  }
  ...
}
```

This would be similar to Apache Spark:

``` scala
case class Subquery(child: LogicalPlan, correlated: Boolean) extends OrderPreservingUnaryNode {
```

Also similar to Calcite:

``` java
public class RexSubQuery extends RexCall {
  public final RelNode rel;
```  

I'm not sure how else we would model this but am open to suggestions.

Reply:
@jackwener @xudong963 @mingmwang @paveltiunov fyi :point_up: this is related to conversations you have all been involved in previously so would like to get your feedback on this PR if possible